core-bundle/tests/Controller/ContentElement/ContentElementControllerTest.php

```diff
@@ -14,22 +14,56 @@
 
 use Contao\ContentModel;
 use Contao\CoreBundle\Fixtures\Controller\ContentElement\TestController;
-use Contao\CoreBundle\Fixtures\Controller\ContentElement\TestSharedMaxAgeController;
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\CoreBundle\Security\Authentication\Token\TokenChecker;
 use Contao\CoreBundle\Tests\TestCase;
 use Contao\FrontendTemplate;
 use Contao\System;
 use FOS\HttpCache\ResponseTagger;
+use ReflectionClass;
 use Symfony\Component\DependencyInjection\ContainerBuilder;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\RequestStack;
 
 class ContentElementControllerTest extends TestCase
 {
+    /**
+     * @var string
+     */
+    private static $testRoot;
+
+    public static function setUpBeforeClass(): void
+    {
+        parent::setUpBeforeClass();
+
+        // Make resources available in test root
+        self::$testRoot = self::getTempDir();
+    }
+
     protected function setUp(): void
     {
         parent::setUp();
 
-        System::setContainer($this->getContainerWithContaoConfiguration());
+        $container = $this->getContainerWithContaoConfiguration();
+        $container->set('request_stack', new RequestStack());
+
+        // Prevent that the dca extractor tries to load language file and dca files
+        // See Contao\DcaExtractor::createExtract()
+        $GLOBALS['TL_LANG']['MSC'] = ['foo' => 'bar'];
+
+        System::setContainer($container);
+    }
+
+    protected function tearDown(): void
+    {
+        parent::tearDown();
+
+        unset($GLOBALS['TL_LANG']['MSC']);
```

Comment:
```suggestion
        unset($GLOBALS['TL_LANG']);
```
Otherwise `$GLOBALS['TL_LANG']` would stay as an empty array.